### PR TITLE
Append workspace name to the runfiles directory name for `pkg_files`

### PR DESCRIPTION
**Pull request by sitaktif**
This is a follow-up to #856, which applies to most rules but `pkg_files` uses a different code path. This change applies the fix to `pkg_files` and adds a test to avoid future regressions.

It also fixes a (silent) bug in the test logic.

### CODE REVIEW

**Comment by sitaktif** [comment]
@aiuto this should probably make it before the next release to avoid confusion.

**Review by cgrindel — APPROVED** [review]

**Comment by sitaktif** [comment]
Thanks @cgrindel - I don't have write access so I'll defer merging to you.

**Comment by cgrindel** [comment]
@aiuto Do you want to look at this before we merge?

**Comment by sitaktif** [comment]
Thanks!

**Pull request merged**
